TST: scipy.misc.ascent has been deprecated in SciPy v1.10.0

Example log: https://github.com/astropy/astropy/runs/8299487818?check_suite_focus=true

```
______________ TestCompressedImage.test_comp_image_quantize_level ______________

self = <astropy.io.fits.tests.test_image.TestCompressedImage object at 0x7fd57406a650>

    @pytest.mark.skipif('not HAS_SCIPY')
    def test_comp_image_quantize_level(self):
        """
        Regression test for https://github.com/astropy/astropy/issues/5969
    
        Test that quantize_level is used.
    
        """
        import scipy.misc
        np.random.seed(42)
>       data = scipy.misc.ascent() + np.random.randn(512, 512)*10

.../astropy/io/fits/tests/test_image.py:1137: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

args = (), kwargs = {}

    @functools.wraps(fun)
    def call(*args, **kwargs):
>       warnings.warn(msg, category=DeprecationWarning,
                      stacklevel=stacklevel)
E       DeprecationWarning: scipy.misc.ascent has been deprecated in SciPy v1.10.0;
and will be completely removed in SciPy v1.12.0. Dataset methods have moved
into the scipy.datasets module. Use scipy.datasets.ascent instead.

.../scipy/_lib/deprecation.py:18: DeprecationWarning
```

--- a/astropy/io/fits/tests/test_image.py
+++ b/astropy/io/fits/tests/test_image.py
@@ -1125,6 +1125,7 @@ class TestCompressedImage(FitsTestCase):
             assert fd[1].header['BITPIX'] == chdu.header['BITPIX']
 
     @pytest.mark.skipif('not HAS_SCIPY')
+    @pytest.mark.remote_data
     def test_comp_image_quantize_level(self):
         """
         Regression test for https://github.com/astropy/astropy/issues/5969
@@ -1132,17 +1133,26 @@ class TestCompressedImage(FitsTestCase):
         Test that quantize_level is used.
 
         """
-        import scipy.misc
+        import scipy
+        from astropy.utils import minversion
+
+        SCIPY_LT_1_10 = not minversion(scipy, '1.10dev0')
         np.random.seed(42)
-        data = scipy.misc.ascent() + np.random.randn(512, 512)*10
+
+        if SCIPY_LT_1_10:
+            import scipy.misc
+            scipy_data = scipy.misc.ascent()
+        else:
+            import scipy.datasets
+            scipy_data = scipy.datasets.ascent()
+
+        data = scipy_data + np.random.randn(512, 512)*10
 
         fits.ImageHDU(data).writeto(self.temp('im1.fits'))
         fits.CompImageHDU(data, compression_type='RICE_1', quantize_method=1,
-                          quantize_level=-1, dither_seed=5)\
-            .writeto(self.temp('im2.fits'))
+                          quantize_level=-1, dither_seed=5).writeto(self.temp('im2.fits'))
         fits.CompImageHDU(data, compression_type='RICE_1', quantize_method=1,
-                          quantize_level=-100, dither_seed=5)\
-            .writeto(self.temp('im3.fits'))
+                          quantize_level=-100, dither_seed=5).writeto(self.temp('im3.fits'))
 
         im1 = fits.getdata(self.temp('im1.fits'))
         im2 = fits.getdata(self.temp('im2.fits'))

--- a/setup.cfg
+++ b/setup.cfg
@@ -70,6 +70,7 @@ test_all =  # Required for testing, plus packages used by particular tests.
     coverage[toml]
     skyfield>=1.20
     sgp4>=2.3
+    pooch
 recommended =
     scipy>=1.3
     matplotlib>=3.1,!=3.4.0,!=3.5.2

--- a/tox.ini
+++ b/tox.ini
@@ -105,7 +105,9 @@ deps =
 # The following indicates which extras_require from setup.cfg will be installed
 extras =
     test: test
+    devdeps: test_all
     recdeps: recommended
+    recdeps: test_all
     alldeps: all
     alldeps: test_all
 

